pkg/cmd/cmd_update.go

```diff
@@ -48,39 +57,163 @@ func KpmUpdate(c *cli.Context, kpmcli *client.KpmClient) error {
 		}
 	}()
 
-	input_paths := c.Args().Slice()
+	pkgInfos := c.Args().Slice()
 
-	pkg_paths := []string{}
-	if len(input_paths) == 0 {
-		pwd, err := os.Getwd()
-		if err != nil {
-			return reporter.NewErrorEvent(reporter.Bug, err, "internal bugs, please contact us to fix it")
-		}
-		pkg_paths = append(pkg_paths, pwd)
-	} else {
-		pkg_paths = input_paths
+	pwd, err := os.Getwd()
+	if err != nil {
+		return reporter.NewErrorEvent(reporter.Bug, err, "internal bugs, please contact us to fix it.")
 	}
 
-	for _, pkg_path := range pkg_paths {
-		kclPkg, err := kpmcli.LoadPkgFromPath(pkg_path)
-		if err != nil {
-			return err
+	fetchNameAndVersion := func(pkgInfo string) (string, string) {
```

Comment:
This function is a bit long, you can split it into smaller functions. Also, will this function only be executed when `kcl mod update`? Will `kcl mod add` also use this function?

Reply:
removed this function, and now using ParseOciPkgNameAndVersion from cmd_add.go